lightning/src/util/macro_logger.rs

```diff
@@ -194,50 +196,172 @@ macro_rules! log_given_level {
+/// fn main() {
+///     log_trace!(logger, "log message {} {}", arg1, arg2);
+///     log_error!(logger, "log message {} {}", arg1, arg2);
+/// }
+/// ```
+/// Example use from external crate:
+/// ```ignore
+/// use lightning::util::default_logging::*;
+/// fn main() {
+///     log_trace!(logger, "log message {} {}", arg1, arg2);
+///     log_error!(logger, "log message {} {}", arg1, arg2);
+/// }
+/// ```
+pub mod default_logging {
+	pub use _log_given_level as log_given_level;
+	pub use _log_debug as log_debug;
+	pub use _log_warn as log_warn;
+	pub use _log_error as log_error;
+	pub use _log_trace as log_trace;
+	pub use _log_info as log_info;
+	pub use _log_gossip as log_gossip;
+}
+
+/// Context Logging
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_given_level_with_context {
+	($logger: expr, $lvl:expr, $peer_id:expr, $channel_id:expr, $($arg:tt)+) => (
+		$crate::__log_given_level!($logger, $lvl, $peer_id, $channel_id, $($arg)*);
+	);
+}
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_error_with_context {
+	($logger: expr, $peer_id:expr, $channel_id:expr, $($arg:tt)*) => (
+		$crate::__log_given_level!($logger, $crate::util::logger::Level::Error, $peer_id, $channel_id, $($arg)*);
+	)
+}
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_warn_with_context {
+	($logger: expr, $peer_id:expr, $channel_id:expr, $($arg:tt)*) => (
+		$crate::__log_given_level!($logger, $crate::util::logger::Level::Warn, $peer_id, $channel_id, $($arg)*);
+	)
+}
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_info_with_context {
+	($logger: expr, $peer_id:expr, $channel_id:expr, $($arg:tt)*) => (
+		$crate::__log_given_level!($logger, $crate::util::logger::Level::Info, $peer_id, $channel_id, $($arg)*);
+	)
+}
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_debug_with_context {
+	($logger: expr, $peer_id:expr, $channel_id:expr, $($arg:tt)*) => (
+		$crate::__log_given_level!($logger, $crate::util::logger::Level::Debug, $peer_id, $channel_id, $($arg)*);
+	)
+}
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_trace_with_context {
+	($logger: expr, $peer_id:expr, $channel_id:expr, $($arg:tt)*) => (
+		$crate::__log_given_level!($logger, $crate::util::logger::Level::Trace, $peer_id, $channel_id, $($arg)*)
 	)
 }
+#[doc(hidden)]
+#[macro_export]
+macro_rules! _log_gossip_with_context {
+	($logger: expr, $peer_id:expr, $channel_id:expr, $($arg:tt)*) => (
+		$crate::__log_given_level!($logger, $crate::util::logger::Level::Gossip, $peer_id, $channel_id, $($arg)*);
+	)
+}
+
+/// Context Logging Module
+///
+/// In some files / modules it is semantically logical for us to include, in the logs, furthur context
+/// of the operating environment, this currently includes the public key of the peer as well as the
+/// channel_id (temporary or 'real').
+/// For example in the case of chain** files, since the channels and peers aren't available / setup,
+/// peer_id and channel_id are irrelevant and Default Logging (above) should be used instead
```

Comment:
> IIUC, we are simply passing the logger to functions from modules that do have that information
yes this is correct. the logger is passed in and is configurable, but _what_ to log for each "object
" (such as channel monitor) is currently defined in the file itself.

currently the importing of context_logging::* means that we expect some "context" from logging in this file (for majority of cases), instead of just logging "anything".

but your question is good, currently the "context" is hard coded to be peer_id and channel_id
and there is no facility to define what happens when we add other kinds of context here. 

imo ideally context should also be determined within files themselves, together with shorthands, (for example for payments related files you might want to always log the payment hash so that you can trace a single payment throughout the logging)

currently that is out of scope of the task but i think its worth exploring more.